imports/plugins/core/catalog/server/hooks/hooks.js

```diff
@@ -0,0 +1,25 @@
+import { Catalog } from "/lib/collections";
+import { Hooks } from "/server/api";
+
+/**
+ * @summary archives product in the Catalog collection when it is archived in Products collection
+ * @param {String} userId - userId of user making the call
+ * @param {Object} doc - product document
+ * @return {undefined}
+ * @private
+ */
+Hooks.Events.add("afterRemoveCatalogProduct", (userId, doc) => {
+  // If this is a top level product, update the `Catalog` collection
+  // to set `isDeleted` to true
+  if (doc.type === "simple") {
+    Catalog.update({
+      "product.productId": doc._id
+    }, {
+      $set: {
+        "product.isDeleted": true
+      }
+    });
+  }
```

Comment:
Did you test toggling visibility/archive of variants and options, without deleting the parent product? Pretty sure I tested that back when we changed the catalog schema, but I just want to make sure that works, too.

Reply:
@aldeed  @kieckhafer  I have verified that visibility and archiving of options works. 👍